code/script.js

```diff
@@ -221,105 +269,174 @@ const generateBoard = () => {
       // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value} hair! Keep all people that have ${value} hair.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
       // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value} hair! Remove all people that have ${value} hair.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// when clicking guess, the player first has to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  const confirmation = confirm(`Are you sure you want to guess ${personToConfirm}?`)
   // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirmation === true) {
+    checkMyGuess(personToConfirm)
+  }
 }
-
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex';
   // 4. Hide the game board
+  board.style.display = 'none';
+  // 2. Set a Message to show in the win or lose section accordingly
+  if (secret.name === personToCheck) {
+    winOrLoseText.innerHTML = `You guessed right! Congratulations, you are a winner!`
+  } else {
+    winOrLoseText.innerHTML = `You lost! The correct person was ${secret.name}.`
+  }
 }
-
 // Invokes the start function when website is loaded
 start()
+  
+// Show the correct alert message for different categories
+alert(`You have ${charactersInPlay.length} characters left.`)
 
 // All the event listeners
 restartButton.addEventListener('click', start)
+
+questions.addEventListener('change', selectQuestion)
+
+findOutButton.addEventListener('click', checkQuestion)
+
+playAgainButton.addEventListener('click', start)
```

Comment:
I see you mentioned some of this in your readme - the play again (after guessing) and the restart (while playting) buttons work well (you have the event listeners set up), however the game does not reset. The scroll down menu should go back to "please choose an option" and timer should start. The secret person does change however. Perhaps the end of my code can help you here - notice the start function being called once more:

//Restart game
restartButton.addEventListener('click', start)
//Play again from the win/ lose site
playAgainButton.addEventListener('click', () => {
  winOrLose.style.display = "none"
  start()